pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -884,11 +912,12 @@ func (oe *operationExecutor) generateUnmountDeviceFunc(
 		unmountDeviceErr := volumeDetacher.UnmountDevice(deviceMountPath)
 		if unmountDeviceErr != nil {
 			// On failure, return error. Caller will log and retry.
-			return fmt.Errorf(
+			err := fmt.Errorf(
 				"UnmountDevice failed for volume %q (spec.Name: %q) with: %v",
 				deviceToDetach.VolumeName,
 				deviceToDetach.VolumeSpec.Name(),
 				unmountDeviceErr)
```

Comment:
@saad-ali - I did add the err variable just in case we use in the future for an event (if we so decide), would you prefer I revert that back to ```return fmt.Errorf(...)```  for this PR?